### ISSUE TITLE
`ClassVar` not respected when indirectly imported

### ISSUE THREAD

**Issue by patrick-kidger** [microsoft/pyright#4965]
```python
# tmp.py
from typing import ClassVar as CV

# tmp2.py
from tmp import CV
from dataclasses import dataclass

@dataclass
class X:
    x: CV[bool]

X()  # Argument missing for parameter "x"
```
Expected behaviour is no error message.

Python 3.8.16, Pyright version 1.1.302 run from the command line.

Related to #3025, I think. It sounds like this is probably a wontfix, so I'm really just opening this bug in response to the "if it comes up frequently enough" message of the previous issue.

**Comment by erictraut** [comment]
Yeah, your assumptions are correct — both in that this is related to to https://github.com/microsoft/pyright/issues/3025 and that it's a "won't fix". This is something that pyright needs to know about when it's building symbol tables, and that's done long before type evaluation is possible.
